kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,159 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Implementation of MPCA->Feature Selection->Linear SVM/LogisticRegression Pipeline
+"""
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.feature_selection import f_classif
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn.utils.validation import check_is_fitted
+
+from ..embed.mpca import MPCA
+
+classifiers = {"svc": [SVC, {"kernel": ["linear"], "C": np.logspace(-3, 2, 6)}],
+               "lr": [LogisticRegression, {"C": np.logspace(-3, 2, 6)}]}
```

Comment:
Again, consider to define repeated values `np.logspace(-3, 2, 6)` as a (global) variable